controllers/helmrepository_controller.go

```diff
@@ -202,7 +204,7 @@ func (r *HelmRepositoryReconciler) reconcile(ctx context.Context, repository sou
 		repository.ObjectMeta.GetObjectMeta(),
 		chartRepo.Index.Generated.Format(time.RFC3339Nano),
 		fmt.Sprintf("index-%s.yaml", url.PathEscape(chartRepo.Index.Generated.Format(time.RFC3339Nano))))
-	if meta.HasReadyCondition(repository.Status.Conditions) && repository.GetArtifact().HasRevision(artifact.Revision) {
+	if meta.InReadyCondition(repository.Status.Conditions) && repository.GetArtifact().HasRevision(artifact.Revision) {
```

Comment:
Given the subtle difference, I wonder if it would be better to do

```suggestion
	if apimeta.IsStatusConditionTrue(repository.Status.Conditions, meta.ReadyCondition) && bucket.GetArtifact().HasRevision(artifact.Revision) {
```

to get rid of another "custom" way of doing things.